pandas/core/generic.py

```diff
@@ -1451,8 +1451,8 @@ def equals(self, other: object) -> bool_t:
         the same location are considered equal.
 
         The row/column index do not need to have the same type, as long
-        as the values are considered equal. Corresponding columns must be of
-        the same dtype.
+        as the values are considered equal. Corresponding columns and
+        index must be of the same dtype.
```

Comment:
I am not sure this is a good clarification? Because now this second sentence is contradicting with the first sentence ("row/column index do not need to have the same dtype") ..

Reply:
It also doesn't seem to be true that the index dtype has to be the same (as also shown in the examples section):

```
In [12]: s1 = pd.Series([1, 2, 3])

In [13]: s2 = pd.Series([1, 2, 3], index=pd.Index([0., 1., 2.]))

In [14]: s1.equals(s2)
Out[14]: True

In [15]: s1.index.dtype
Out[15]: dtype('int64')

In [16]: s2.index.dtype
Out[16]: dtype('float64')
```